footylib/footylib.py

```diff
@@ -197,71 +202,59 @@ def _populate(self, info, division):
             self.referee = info.find('td', {'class': 'ref'}).text
             self.motm = info.find('td', {'class': 'man'}).text
             self.datetime = self.__string_to_datetime(self.date, self.time)
-            self._calendar = None
             self.division = division or ''
         except KeyError:
             self.logger.exception("Got an exception on Matches.")
 
-
     @property
     def calendar(self):
         if not self._calendar:
-            cal = FootyCalendar(self.date, self.name, self.location)
+            cal = FootyCalendar(self.datetime, self.teams, self.location)
             self._calendar = cal.generate()
+            self.logger.debug(self._calendar)
         return self._calendar
 
     @staticmethod
     def __string_to_datetime(date, time):
-        months = {"januari": "January",
-                  "februari": "February",
-                  "maart": "March",
-                  "april": "April",
-                  "mei": "May",
-                  "juni": "June",
-                  "juli": "July",
-                  "augustus": "August",
-                  "september": "September",
-                  "oktober": "October",
-                  "november": "November",
-                  "december": "December"}
-        dutch_datetime = '{} {}'.format(date, time).split()
-        # TODO: look for NL locale
-        english_datetime = months[dutch_datetime[0]]
-        dutch_datetime[0] = english_datetime
-        english_datetime = " ".join(dutch_datetime)
+        locale.setlocale(locale.LC_TIME, 'nl_NL')
+        dutch_datetime = '{} {}'.format(date.capitalize(), time)
         try:
-            datetime_object = datetime.strptime(english_datetime,
+            datetime_object = datetime.strptime(dutch_datetime,
                                                 '%B %d, %Y %I:%M %p')
             return datetime_object
         except ValueError:
             LOGGER.exception("Couldn't parse this datetime.")
 
 
 class FootyCalendar(object):
-    def __init__(self, season):
+    def __init__(self, datetime, teams, location):
```

Comment:
Here you probably broke more stuff than you can realize by overloading the datetime module. This is very difficult to debug.